lib/pbench/server/api/resources/users_api.py

```diff
@@ -284,24 +285,55 @@ def post(self):
-        # Expired token is already deleted by now if we reach here.
-        # In either case we will return suceess to a client, since the same token can not be used again.
+        # The only time a token won't be in the database is if the token is
+        # not a valid token.  Valid tokens that are not expired should be in
+        # the database, and expired tokens might or might not be in the
+        # database.  If the valid token is found in the database associated
+        # with the user, remove it.
 
-        if user:
-            try:
-                ActiveTokens.delete(auth_token)
-            except Exception:
-                current_app.logger.exception(
-                    "Exception occurred deleting an auth token"
-                )
-                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-            else:
-                current_app.logger.debug("User {} logged out", user.username)
+        if state == Auth.TokenState.INVALID:
+            current_app.logger.info("User logout with invalid token: {}", auth_token)
         else:
-            current_app.logger.info("User logout with invalid or expired token")
+            token = ActiveTokens.query(auth_token)
+            if token:
+                user = token.user
+                if not user:
+                    # This is an internal error state, but play it cool with the
+                    # API client.
+                    current_app.logger.error(
+                        "INTERNAL ERROR - token {!r} ({}) found without an associated user",
+                        auth_token,
+                        state,
+                    )
+                else:
+                    try:
+                        ActiveTokens.delete(auth_token)
+                    except Exception:
+                        current_app.logger.exception(
+                            "Exception occurred deleting auth token {!r} for user {!r}",
+                            auth_token,
+                            user.username,
+                        )
+                        if state == Auth.TokenState.VERIFIED:
+                            # Only report an internal error to the user if the token
+                            # has been verified.
+                            abort(
+                                HTTPStatus.INTERNAL_SERVER_ERROR,
+                                message="INTERNAL ERROR",
+                            )
+                    else:
+                        if state == Auth.TokenState.EXPIRED:
+                            current_app.logger.info(
+                                "User {} logged out with expired token", user.username
+                            )
+                        else:
+                            current_app.logger.debug(
+                                "User {} logged out", user.username
+                            )
```

Comment:
Now that this change is "done", are we sure that we want to log these?  Logging out with an expired token probably isn't weird.  And, logout should be an audit event, and it's a user action, so should it really be logged (above the `debug` level)?